tests/unit/test_steps.py

```diff
@@ -372,6 +367,40 @@ def test_chaining_steps():
     assert s1.next_step == s2
     assert s2.next_step == s3
 
+
+def test_chaining_choice():
+    s1_pass = Pass('Step - One')
+    s2_choice = Choice('Step - Two')
+    s3_pass = Pass('Step - Three')
+
+    chain1 = Chain([s1_pass, s2_choice, s3_pass])
+    assert chain1.steps == [s1_pass, s2_choice, s3_pass]
+    assert s1_pass.next_step == s2_choice
+    assert s2_choice.default == s3_pass
+    assert s2_choice.next_step is None  # Choice steps do not have next_step
+    assert s3_pass.next_step is None
+
+
+def test_chaining_choice_with_default(caplog):
+    s1_pass = Pass('Step - One')
+    s2_choice = Choice('Step - Two')
+    s3_pass = Pass('Step - Three')
+
+    s2_choice.default_choice(s3_pass)
+
+    # Chain s2_choice when default_choice is already set will trigger Warning
+    with caplog.at_level(logging.WARNING):
```

Comment:
Seems like overkill to make assertion son logging, but I appreciate the thoroughness 😄